pkcs11/signer.go

```diff
@@ -285,13 +289,11 @@ func getX509CACert(ctx context.Context, key config.KeyConfig, pool sPool, hostna
 	return cert, nil
 }
 
-func getSignatureAlgorithm(pka crypki.PublicKeyAlgorithm) x509.SignatureAlgorithm {
-	switch pka {
-	case crypki.RSA:
-		return x509.SHA256WithRSA
-	case crypki.ECDSA:
-		return x509.ECDSAWithSHA256
-	default:
-		return x509.SHA256WithRSA
+func isValidCertRequest(cert *x509.Certificate, sa crypki.SignatureAlgorithm) bool {
```

Comment:
nit: can we make it a one liner?
`return cert.SignatureAlgorithm == x509cert.GetSignatureAlgorithm(sa)` ?